packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -0,0 +1,895 @@
+        require(support <= 2, 'NounsDAO::castVoteInternal: invalid vote type');
+        Proposal storage proposal = _proposals[proposalId];
+        Receipt storage receipt = proposal.receipts[voter];
+        require(receipt.hasVoted == false, 'NounsDAO::castVoteInternal: voter already voted');
+
+        /// @notice: Unlike GovernerBravo, votes are considered from the block the proposal was created in order to normalize quorumVotes and proposalThreshold metrics
+        uint96 votes = nouns.getPriorVotes(voter, proposalCreationBlock(proposal));
+
+        if (support == 0) {
+            proposal.againstVotes = proposal.againstVotes + votes;
+        } else if (support == 1) {
+            proposal.forVotes = proposal.forVotes + votes;
+        } else if (support == 2) {
+            proposal.abstainVotes = proposal.abstainVotes + votes;
+        }
+
+        receipt.hasVoted = true;
+        receipt.support = support;
+        receipt.votes = votes;
+
+        return votes;
+    }
+
+    /**
+     * @notice Admin function for setting the voting delay
+     * @param newVotingDelay new voting delay, in blocks
+     */
+    function _setVotingDelay(uint256 newVotingDelay) external {
+        require(msg.sender == admin, 'NounsDAO::_setVotingDelay: admin only');
+        require(
+            newVotingDelay >= MIN_VOTING_DELAY && newVotingDelay <= MAX_VOTING_DELAY,
+            'NounsDAO::_setVotingDelay: invalid voting delay'
+        );
+        uint256 oldVotingDelay = votingDelay;
+        votingDelay = newVotingDelay;
+
+        emit VotingDelaySet(oldVotingDelay, votingDelay);
+    }
+
+    /**
+     * @notice Admin function for setting the voting period
+     * @param newVotingPeriod new voting period, in blocks
+     */
+    function _setVotingPeriod(uint256 newVotingPeriod) external {
+        require(msg.sender == admin, 'NounsDAO::_setVotingPeriod: admin only');
+        require(
+            newVotingPeriod >= MIN_VOTING_PERIOD && newVotingPeriod <= MAX_VOTING_PERIOD,
+            'NounsDAO::_setVotingPeriod: invalid voting period'
+        );
+        uint256 oldVotingPeriod = votingPeriod;
+        votingPeriod = newVotingPeriod;
+
+        emit VotingPeriodSet(oldVotingPeriod, votingPeriod);
+    }
+
+    /**
+     * @notice Admin function for setting the proposal threshold basis points
+     * @dev newProposalThresholdBPS must be greater than the hardcoded min
+     * @param newProposalThresholdBPS new proposal threshold
+     */
+    function _setProposalThresholdBPS(uint256 newProposalThresholdBPS) external {
+        require(msg.sender == admin, 'NounsDAO::_setProposalThresholdBPS: admin only');
+        require(
+            newProposalThresholdBPS >= MIN_PROPOSAL_THRESHOLD_BPS &&
+                newProposalThresholdBPS <= MAX_PROPOSAL_THRESHOLD_BPS,
+            'NounsDAO::_setProposalThreshold: invalid proposal threshold'
```

Comment:
Leftover from V1, but could you update to `NounsDAO:: _setProposalThresholdBPS: invalid proposal threshold bps`

Reply:
fixed: https://github.com/nounsDAO/nouns-monorepo/pull/438/commits/605e2f61d272508affc85ecc33a8d2b28ee14295